packages/app/src/app/pages/Dashboard/Components/Branch/BranchCard.tsx

```diff
@@ -18,117 +17,86 @@ export const BranchCard: React.FC<BranchProps> = ({
-          paddingY: 11,
-          position: 'relative',
-        })}
-        align="center"
-        justify="center"
-      >
-        <Icon color="#808080" name="branch" size={32} />
-      </Stack>
-      {restricted ? (
-        <Element css={{ position: 'absolute', top: 8, left: 8 }}>
-          <Badge variant="trial">Restricted</Badge>
-        </Element>
-      ) : null}
-      <Stack
-        css={css({
-          paddingY: 5,
-          paddingLeft: 5,
-          paddingRight: 2,
-          justifyContent: 'space-between',
-          height: '100%',
-        })}
-        direction="vertical"
-      >
-        <Stack align="center" justify="space-between">
-          <Tooltip label={branchName}>
-            <Text
-              css={css({
-                color: restricted ? '#999999' : '#E5E5E5',
-                flex: 1,
-                whiteSpace: 'nowrap',
-                overflow: 'hidden',
-                textOverflow: 'ellipsis',
-              })}
-              weight="medium"
-              size={13}
-            >
-              {branchName}
-            </Text>
-          </Tooltip>
-          <IconButton
-            variant="square"
-            name="more"
-            size={14}
-            title="Branch actions"
-            onClick={evt => {
-              evt.stopPropagation();
-              onContextMenu(evt);
-            }}
-          />
-        </Stack>
-        <Stack gap={2}>
-          {contribution && (
-            <Icon color="#EDFFA5" name="contribution" size={16} />
-          )}
-          <Text
-            css={css({
-              color: '#808080',
-              flex: 1,
-              whiteSpace: 'nowrap',
-              overflow: 'hidden',
-              textOverflow: 'ellipsis',
-            })}
-            size={13}
-          >
-            {repository.owner}/{repository.name}
-          </Text>
+    <InteractiveOverlay>
+      <StyledCard dimmed={isBeingRemoved}>
+        <Stack
+          css={{ height: '100%' }}
+          direction="vertical"
+          justify="space-between"
+        >
+          <Stack justify="space-between">
+            <Stack direction="vertical" gap={1} css={{ overflow: 'hidden' }}>
+              {showRepo && (
+                <Text
+                  css={{
+                    color: '#808080',
+                    whiteSpace: 'nowrap',
+                    overflow: 'hidden',
+                    textOverflow: 'ellipsis',
+                  }}
+                  size={12}
+                >
+                  {repository.owner}/{repository.name}
+                </Text>
+              )}
+
+              <Stack gap={2}>
+                {contribution ? (
+                  <Icon color="#EDFFA5" name="contribution" size={16} />
+                ) : (
+                  <Icon color="#999999" name="branch" size={16} />
+                )}
+                <InteractiveOverlay.Item
+                  as="a"
+                  aria-label={ariaLabel}
```

Comment:
I wonder if we should use a visually hidden label and add `aria-hidden` around the `branchName` instead of using `aria-label`. The main reason to do this would be [a11y improvement for translation](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/). This would mean that we also need to [wrap the branch and repo names in elements with `translate="no"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/translate) though 😄.

Reply:
I don't know if it's worth doing this right now, but it is a great idea for adapting this to the new codebase. Maybe we can add this to our guidelines? https://www.notion.so/codesandbox-app/JS-and-TS-Coding-Guidelines-6d9af67651c740a8ab97746ece59a484 we can create a separate section on accessibility

Reply:
That guidelines doc is long looking for some love, hope to get some time for it in the first part of the year